api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/Procedure.java

```diff
@@ -177,4 +167,47 @@ public String getOutputPath() {
 		return "";
 	}
 
+	public boolean hasPermissions(UserAccount user) {
+		if (isPublicAccessible()) {
+			return true;
+		}
+		if (user == null) {
+			return false;
+		}
+		if (user.isRootUser()) {
+			return true;
+		}
+		for (AccessWhitelist filter : accessWhitelists) {
+			if (filter.isAuthorized(user)) {
```

Comment:
Can we pass here the procedure name? it is strange that we have procedure name in all AccessWhitelist classes just for purposes of logging message

Reply:
Yes.